kotlinx-coroutines-core/common/src/channels/ConflatedChannel.kt

```diff
@@ -24,31 +24,20 @@ internal open class ConflatedChannel<E> : AbstractChannel<E>() {
     protected final override val isBufferAlwaysFull: Boolean get() = false
     protected final override val isBufferFull: Boolean get() = false
 
-    override fun onClosedIdempotent(closed: LockFreeLinkedListNode) {
```

Comment:
This change makes the channel non-linearizable (see `ChannelLinearizabilityTest.testConflatedChannelLinearizability`).

`onClosedIdempotent ` call should conflate all sent elements except the last one.
Otherwise, weird non-linearizable executions are possible when closing thread interacts with the channel after `close`.

Reply:
Yes hell, indeed.